tests/unit/interceptors/http.test.ts

```diff
@@ -28,7 +28,22 @@ describe('HTTP Interceptors', () => {
       await interceptor({ request });
 
       // Assert
-      expect(request.headers.get('Content-Type')).toBe('application/json');
+      expect(request.headers.get('Content-Type')).toBe('text/plain');
+    });
+
+    it('should perform case insensitive checks on headers', async () => {
+      // Arrange
+      const request = new Request('https://example.com', {
+        headers: { 'content-type': 'text/plain' },
+      });
+
+      const interceptor = HttpInterceptors.setDefaultHeaders();
+
+      // Act
+      await interceptor({ request });
+
+      // Assert
+      expect(request.headers.get('Content-Type')).toBe('text/plain');
```

Comment:
I'm a bit confused by this. Reading `src/interceptors/http.ts` I'd expect this to be set to `application/json` from the default headers as (lowercase) `content-type` is not in the map `DEFAULT_HEADERS`?

Reply:
Two things basically:
- Headers’ names are checked case insensitive. This is achieved through the HTTP Headers instance's methods (.get/.has) that already handle this for us. 
- Headers should default to `application/json` only if the user has not manually overridden the value (which is the case here), hence the return value

Reply:
Makes sense thanks!